meta-mbl-distro/recipes-extended/shadow/shadow-sysroot_%.bbappend

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+MBL_MIN_PASS_LEN ?= "5"
+
+do_install_append () {
+    # We want to use SHA512 as encryption method to generate passwords for the target
+    sed -i 's/^#ENCRYPT_METHOD.*$/ENCRYPT_METHOD SHA512/' ${D}${sysconfdir}/login.defs
+
+    # We want the minimum password length set by MBL_MIN_PASS_LEN for the target
+    sed -i 's/^PASS_MIN_LEN.*$/PASS_MIN_LEN ${@d.getVar('MBL_MIN_PASS_LEN', True)}/' ${D}${sysconfdir}/login.defs
```

Comment:
Why are you using Python to get the value of `MBL_MIN_PASS_LEN` rather than just `${MBL_MIN_PASS_LEN}`?

Reply:
I'm confused. This code seems to imply that login.defs already contains a `PASS_MIN_LEN` setting, but we haven't added it in this PR. If the shadow packages already support a `PASS_MIN_LEN` setting then why aren't they already checking it in {user,group}{add,mod}? If they already have this setting then why is the upstreaming status "Inappropriate [OE specific]" - shouldn't they be using it (like your patch does)?

Reply:
The upstream doesn't support plain text password and it was added for oe-core via this patch:
https://github.com/openembedded/openembedded-core/blob/master/meta/recipes-extended/shadow/shadow.inc#L28

In the upstream the `PASS_MIN_LEN` is only checked when using or `passwad` commands VIA `new_password` function call.

I see a typo in the `MBL_MIN_PASS_LEN`, as it should be `MBL_PASS_MIN_LEN`.

Reply:
Ah, okay, that explains it... So should the patches go in a separate bbappend bbappend in oe-core-mbl and get upstreamed to oe-core?

Reply:
FWIW, I'm a bit nervous about these patches anyway. If we can't upstream them then they seem like a pretty big maintenance burden.

Reply:
I can move this patch to openembedded-core-mbl and we can try to upstream it.
I just like to mention that this patch is only applied for the native package, anyway.

Reply:
> I just like to mention that this patch is only applied for the native package, anyway.

Huh? So what's the point then? We've already got checks at build time that the root password is long enough.

Reply:
Why isn't the patch applied for the target package?

Reply:
> Why isn't the patch applied for the target package?

Because in the target it is not supposed to support plain text passwords.

Reply:
> > I just like to mention that this patch is only applied for the native package, anyway.
> 
> Huh? So what's the point then? We've already got checks at build time that the root password is long enough.

Ok, I'll remove it.

Reply:
Oh, right, good :-)

I suggest we just get rid of the patches then. I don't think they're worth the maintenance burden.

Reply:
Addressed in 4a3571b.